src/app.ts

```diff
@@ -114,23 +105,35 @@ const samlConfig: SamlConfig = {
 };
 
 const acs: AssertionConsumerServiceT = async user => {
-  /* Should validate the user */
-  // Create token
-  const token = crypto.randomBytes(32).toString("hex");
-
-  // Write token in redis
-  redisClient.set(
-    `session-token:${token}`,
-    JSON.stringify(user),
-    "EX",
-    3600, // expire sec 1 h
-    (_, __) => new Error("Error setting session token")
-  );
-
-  // Add token in query string
-  return ResponsePermanentRedirect({
-    href: `${config.ENDPOINT_SUCCESS}?token=${token}`
-  });
+  return fromEither(SpidUser.decode(user))
+    .mapLeft(errorsToError)
+    .chain(spidUser =>
+      config.ENABLE_JWT
+        ? getSpidUserJwt(
+            config.JWT_TOKEN_PRIVATE_RSA_KEY,
+            spidUser,
+            config.JWT_TOKEN_EXPIRATION,
+            config.JWT_TOKEN_ISSUER
+          )
+        : taskEither
+            .of<Error, string>(crypto.randomBytes(32).toString("hex"))
+            .chain(token =>
+              setWithExpirationTask(
+                redisClient,
+                `session-token:${token}`,
```

Comment:
Can we define a global constat with the key prefix?

Reply:
Done in [54fbbec](https://github.com/pagopa/hub-spid-login-ms/pull/1/commits/54fbbecb2348fd01818728a356eb39516ca32125)